Review.swift

```diff
@@ -0,0 +1,294 @@
+                            comments.append(Comment(id: nil,
+                                                    line: nil,
+                                                    lineType: "",
+                                                    ruleDescription: generalComment.ruleDescription,
+                                                    content: generalComment.reason,
+                                                    path: "",
+                                                    type: CommentType(rawValue: generalComment.severity.rawValue)))
+                        }
+                        
+                        if !comments.isEmpty {
+                            logger.info("Posting General Comments...")
+                            sourceControlService.postGeneralComment(repositoryName: repository.name, pullRequest: pullRequest, comments: comments, req: req)
+                        }
+                        
+                        if generalComments.count == 0 && inlineComments.count == 0 {
+                            logger.info("Approving Pull Request...")
+                            sourceControlService.approvePullRequest(repositoryName: repository.name, pullRequestId: pullRequest.scmId, req: req)
+                        } else {
+                            logger.info("Pull Request Needs Work...")
+                            sourceControlService.markNeedsRework(repositoryName: repository.name, pullRequestId: pullRequest.scmId, req: req).whenSuccess { _ in
+                                logger.info("Generating Tasks...")
+                                let newTasks = self.generateTasks(for: violations)
+                                
+                                if !formerTasks.isEmpty {
+                                    logger.info("Resolving former tasks...")
+                                    formerTasks.map { task in
+                                        
+                                        guard let task = task else {
+                                            logger.warning("Task could not be unwrapped!")
+                                            return
+                                        }
+                                        
+                                        guard let id = task.id else {
+                                            logger.warning("Task ID could not be unwrapped!")
+                                            return
+                                        }
+                                        
+                                        sourceControlService.resolveTask(id: id, req: req)
+                                    }
+                                }
+                                
+                                logger.info("Posting Tasks...")
+                                sourceControlService.postTasks(repositoryName: repository.name, pullRequestId: pullRequest.scmId, tasks: newTasks, req: req).whenSuccess { status in
```

Comment:
🚫 Error
 [Line Length] At line 126: Line should be 150 characters or less: currently 180 characters